javaparser-core/src/main/java/com/github/javaparser/CommentsInserter.java

```diff
@@ -46,59 +43,46 @@
     CommentsInserter(ParserConfiguration configuration) {
         this.configuration = configuration;
     }
-
-    /**
-     * Adds the comments found in the source code of a compilation unit to that compilation unit.
-     * @param cu an already created compilation unit
-     * @param cuSourceCode the source code of the compilation unit. It will be parsed to find comments.
-     */
-    public void insertComments(CompilationUnit cu, String cuSourceCode) throws IOException {
-        CommentsParser commentsParser = new CommentsParser();
-        CommentsCollection allComments = commentsParser.parse(cuSourceCode);
-
-        insertCommentsInCu(cu, allComments);
-    }
-
+    
     /**
-     * Comments are attributed to the thing the comment and are removed from
-     * allComments.
+     * Comments are attributed to the thing they comment and are removed from
+     * the commentsCollection.
      */
-    private void insertCommentsInCu(CompilationUnit cu,
-            CommentsCollection commentsCollection) {
-        if (commentsCollection.size() == 0)
+    private void insertComments(CompilationUnit cu, TreeSet<Comment> comments) {
+        if (comments.isEmpty())
             return;
 
-        // I should sort all the direct children and the comments, if a comment
-        // is the first thing then it
-        // a comment to the CompilationUnit
-        // FIXME if there is no package it could be also a comment to the
-        // following class...
+        /* I should sort all the direct children and the comments, if a comment
```

Comment:
Because the comment spans multiple lines and was getting wrecked in reformatting. A block comment should keep the content of the comment together better.